internal/common/common_test.go

```diff
@@ -678,3 +688,415 @@ func TestNewSyncableSecretMetaData(t *testing.T) {
+					Name:      "foo",
+					Namespace: "baz",
+				},
+				Spec: secretsv1beta1.VaultAuthSpec{
+					VaultAuthGlobalRef: "buz",
+					Method:             "gcp",
+				},
+			},
+			gObj: gObj.DeepCopy(),
+			want: &secretsv1beta1.VaultAuth{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "foo",
+					Namespace:       "baz",
+					ResourceVersion: "1",
+				},
+				Spec: secretsv1beta1.VaultAuthSpec{
+					VaultConnectionRef: "default",
+					VaultAuthGlobalRef: "buz",
+					Method:             "gcp",
+					Namespace:          "biff",
+					Mount:              "qux",
+					GCP: &secretsv1beta1.VaultAuthConfigGCP{
+						Role:                           "beetle",
+						Region:                         "us-west1",
+						WorkloadIdentityServiceAccount: "sa1",
+					},
+				},
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "global-ref-not-set",
+			c:    builder.Build(),
+			o: &secretsv1beta1.VaultAuth{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: "baz",
+				},
+				Spec: secretsv1beta1.VaultAuthSpec{},
+			},
+			want: &secretsv1beta1.VaultAuth{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:            "foo",
+					Namespace:       "baz",
+					ResourceVersion: "1",
+				},
+				Spec: secretsv1beta1.VaultAuthSpec{},
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "invalid-method",
+			c:    builder.Build(),
+			o: &secretsv1beta1.VaultAuth{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: "baz",
+				},
+				Spec: secretsv1beta1.VaultAuthSpec{
+					VaultAuthGlobalRef: "buz",
+					Method:             "invalid",
+				},
+			},
+			gObj: gObj.DeepCopy(),
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				return assert.EqualError(t, err,
+					`unsupported auth method "invalid" for global auth merge`)
+			},
+		},
+		{
+			name: "invalid-global-ref",
+			c:    builder.Build(),
+			o: &secretsv1beta1.VaultAuth{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "foo",
+					Namespace: "baz",
+				},
+				Spec: secretsv1beta1.VaultAuthSpec{
+					VaultAuthGlobalRef: "invalid",
+				},
+			},
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				if assert.ErrorContains(t, err, "failed getting baz/invalid, err=") {
+					return assert.True(t, errors.IsNotFound(err), i...)
+				}
+				return false
+			},
+		},
+		{
+			name: "invalid-nil-auth-config",
+			c:    builder.Build(),
+			gObj: &secretsv1beta1.VaultAuthGlobal{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "baz",
+					Name:      "buz",
+				},
+			},
+			o: &secretsv1beta1.VaultAuth{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "baz",
+					Name:      "foo",
+				},
+				Spec: secretsv1beta1.VaultAuthSpec{
+					VaultAuthGlobalRef: "buz",
+				},
+			},
+			wantErr: func(t assert.TestingT, err error, i ...interface{}) bool {
+				return assert.ErrorContains(t, err, "no auth method set in VaultAuth baz/foo")
+			},
+		},
```

Comment:
Could we have more test coverages for other auth methods for invalid auth config cases after merged with VaultAuthGlobal?

Reply:
Seems reasonable